examples/sssp/sssp.cu

```diff
@@ -1,44 +1,68 @@
 #include <cstdlib>  // EXIT_SUCCESS
 
-#include <gunrock/applications/sssp/sssp.hxx>
+#include <gunrock/applications/runner.hxx>
+#include <gunrock/applications/sssp/sssp_implementation.hxx>
 
 using namespace gunrock;
 using namespace memory;
 
 void test_sssp(int num_arguments, char** argument_array) {
-  using vertex_t = int;
-  using edge_t = int;
-  using weight_t = float;
-
+  
   if (num_arguments != 2) {
-    std::cerr << "usage: ./bin/color filename.mtx" << std::endl;
+    std::cerr << "usage: ./bin/<program-name> filename.mtx" << std::endl;
     exit(1);
   }
-
-  // Load Matrix-Market file & convert the resultant COO into CSR format.
+  
+  // --
+  // Define types
+  
+  using vertex_t = int;
+  using edge_t   = int;
+  using weight_t = float;
+  
+  // --
+  // IO
+  
   std::string filename = argument_array[1];
+  
   io::matrix_market_t<vertex_t, edge_t, weight_t> mm;
   auto coo = mm.load(filename);
+  
+  format::csr_t<memory::memory_space_t::device, vertex_t, edge_t, weight_t> csr;
+  csr = coo; 
+  // ^^ Honestly don't love the operator overloading here -- it feels unexpected.  
+  // I'd prefer `csr.load(coo)` or, even better, `csr = coo.tocsr()` or `csr = to_csr(coo)`
 
-  // convert coo to csr
-  format::csr_t<memory_space_t::device, vertex_t, edge_t, weight_t> csr;
-  csr = coo;  // Able to convert host-based coo_t to device-based csr (or host
-              // to host). As of right now, it requires coo to be host side.
+  // --
+  // Build graph + metadata
+  
+  auto G    = graph::build::from_csr_t<memory_space_t::device>(&csr);
+  auto meta = graph::build::meta_from_csr_t(&csr);
```

Comment:
I was thinking, what if the build graph always returned the graph `G` and meta `M`, we can write it really cleanly in C++ (I am not sure if we need 17 for this):
```cpp
auto [G, meta] = graph::build::from_csr_t<memory_space_t::device>(&csr);
```

Where the return statement would look like:
```cpp
return {G, meta};
```

Reply:
agree

Reply:
Another note will be hiding `csr_t` behind templates as well, so, it will just be:
```
auto [G, meta] = graph::build<memory_space_t::device>(&csr);
```

Reply:
Not able to get this working -- maybe not supported by 14?  Or maybe doesn't support `auto` return types?

Reply:
There's no reason for me to use `14` if we can do more with `17`. I was just trying to see what I can get away with by staying on `14`. See this line: https://github.com/bkj/essentials/blob/master/examples/Makefile.inc#L70 & https://github.com/bkj/essentials/blob/master/examples/Makefile.inc#L74